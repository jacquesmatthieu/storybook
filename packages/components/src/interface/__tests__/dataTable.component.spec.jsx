import React from 'react';
import { shallow } from 'enzyme';

import { DataTable, DataTableCell } from '../dataTable.component';

describe('DataTable', () => {
  const getWrapper = () => shallow(<DataTable>Foo</DataTable>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DataTableCell', () => {
  let props;

  beforeEach(() => {
    props = {};
  });

  const getWrapper = () => shallow(<DataTableCell {...props}>Bar</DataTableCell>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render head version', () => {
    // Given
    props.head = true;
    props.inputProps = { foo: 'bar' };

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
