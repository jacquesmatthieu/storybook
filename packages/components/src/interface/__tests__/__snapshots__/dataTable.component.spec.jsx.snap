// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should render properly 1`] = `
<dataTablecomponent__WrapperDataTable>
  <dataTablecomponent__StyledDataTable>
    Foo
  </dataTablecomponent__StyledDataTable>
</dataTablecomponent__WrapperDataTable>
`;

exports[`DataTableCell should render head version 1`] = `
<dataTablecomponent__StyledCell>
  <dataTablecomponent__CellContent
    inputProps={
      Object {
        "foo": "bar",
      }
    }
    variant="dataTable"
  >
    Bar
  </dataTablecomponent__CellContent>
</dataTablecomponent__StyledCell>
`;

exports[`DataTableCell should render properly 1`] = `
<dataTablecomponent__StyledCell>
  <dataTablecomponent__CellContent
    variant="dataTable"
  >
    Bar
  </dataTablecomponent__CellContent>
</dataTablecomponent__StyledCell>
`;
